recipes/rgbcx/all/test_package/test_package.cpp

```diff
@@ -0,0 +1,8 @@
+#define RGBCX_IMPLEMENTATION
```

Comment:
Why don't you export it by cpp_info.defines ?

Reply:
Agree. If this is a de facto definition, it should be added to the recipe with `self.cpp_info.defines.append("RGBCX_IMPLEMENTATION")` to avoid manual definition in the consumers

Reply:
`RGBCX_IMPLEMENTATION` must be defined in a single translation unit including rgbcx. Defining it in `cpp_info.defines` can lead to several linking errors (duplicate implementations, ...) because `RGBCX_IMPLEMENTATION` will be defined in all translation units. sdf, tinyexr and stb work like this.


https://github.com/richgel999/bc7enc/blob/master/rgbcx.h#L11

Reply:
thank you for the details. Let's move forward then!